src/server/cluster/cluster_family.cc

```diff
@@ -25,6 +25,9 @@
 #include "server/server_state.h"
 
 ABSL_FLAG(std::string, cluster_announce_ip, "", "ip that cluster commands announce to the client");
+ABSL_FLAG(std::string, cluster_id, "",
```

Comment:
Did you check if in redis is the node id the same as the cluster node id? 
Lets add it to feature tests of cluster i.e
we should have a test for cluster running with replicas and sending data via seeder (if not we need to add), in this test one of the configurations to run is with this flag

Reply:
and also is there a reason not to set this id for the master_replid_ but have different ids

Reply:
suppose we always ` master_replid_` to constant `cluster_node_id`, then in case a master restarts, it will still have the same id and the replica will happily replicate empty data from it. that's what we are trying to prevent, right?

Reply:
> Did you check if in redis is the node id the same as the cluster node id?

No, it's not the same ID. Here's a log from a local redis cluster I set:

```
177592:M 07 Mar 2024 13:01:29.548 * No cluster configuration found, I'm 09a6bad961e78159056f8ad3865a3e2d47d74045
177592:M 07 Mar 2024 13:01:29.560 * Server initialized
[...]
177592:M 07 Mar 2024 13:02:15.440 * Replica 127.0.0.1:6379 asks for synchronization
177592:M 07 Mar 2024 13:02:15.440 * Full resync requested by replica 127.0.0.1:6379
177592:M 07 Mar 2024 13:02:15.440 * Replication backlog created, my new replication IDs are '67e4e538e2e0a76da7a1454615c5ce64954c4800' and '0000000000000000000000000000000000000000'
```

So it's not using the same ID.

> Lets add it to feature tests of cluster i.e we should have a test for cluster running with replicas and sending data via seeder (if not we need to add), in this test one of the configurations to run is with this flag

Done.